### PR TITLE
Input form Testing

### PR DESCRIPTION
**Pull request by rachelwong**
Tests not passing for input form for
- unable to test for colour of notification
- unable to test accept and score dummy word

Whether test accept and score dummy word should be in a separate spec file.

### CODE REVIEW

File: cypress/integration/init.spec.js [hunk 1]
```diff
@@ -21,43 +21,49 @@ describe("Initialise new game instance", () => {
   })
 
   it("Show welcome message", () => {
-    // Assert - welcome msg & UI are visible. And new game button should not be visible
+    // Assert - Welcome msg & UI are visible. And new game button should not be visible
+    // Assert - welcome msg is blue
+    cy.get('#alertBox') // cy-data attri didn't work 
+      .and("have.class", "alert-info")
+    // .should("have.css", "background-color") // checking visually for colour did not compute 
+    // .and('eq', 'rgb(209, 236, 241)')
```
Comment (dan-dekel) [inline cypress/integration/init.spec.js]: We prefer to remove commented out code. You'll find that you'll rarely use code which has been commented out in the past

File: cypress/integration/init.spec.js [hunk 2]
```diff
@@ -21,43 +21,49 @@ describe("Initialise new game instance", () => {
...
-  it("Current game score is 0", () => {
-    // arrange - visit page
-    // action - ui available
-
-    // assert - display current game score as 0
-    cy.get('[cy-data="totalScore"]')
-      .should("contain", 0)
+      .and('contain', 0)
+      .and(() => {
+        expect(sessionStorage.getItem('currentScore')).to.exist
```
Comment (dan-dekel) [inline cypress/integration/init.spec.js]: in cases where we call the same method twice, we could instead call it once:

const actualScore = sessionStorage.getItem('currentScore');
// and then
expect(actualScore ).to.exist
expect(actualScore).to.contain(0)

File: cypress/integration/init.spec.js [hunk 3]
```diff
@@ -21,43 +21,49 @@ describe("Initialise new game instance", () => {
...
+        expect(sessionStorage.getItem('currentScore')).to.contain(0) // to.equal does not work here!
```
Comment (dan-dekel) [inline cypress/integration/init.spec.js]: You are correct, to.equal won't work. I'd like you to find out why. Here's a hint:
https://stackoverflow.com/a/33952336

I'd like you to change your code such that you can use to.equal (you have used a similar method in the script.js file)

File: cypress/integration/inputForm.spec.js [hunk 4]
```diff
@@ -0,0 +1,139 @@
+/// <reference types="Cypress" />
+
+describe("Input Form", () => {
+
+  beforeEach(() => {
+    cy.startGame()
+  })
+
+  it("Accepts input", () => {
+    const dummy = "HELLO"
```
Comment (dan-dekel) [inline cypress/integration/inputForm.spec.js]: for naming variables we can use the intent of that variable. i.e. in this example, you could name it something like "expectedValue" or "expected"

File: cypress/integration/inputForm.spec.js [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+      // assert - see error message: you must only Alphabets only, no numbers or symbols.
+      cy.get('[cy-data="alertTextDisplay"]')
+        .should("be.visible")
+        .contains("Alphabets only, no numbers or symbols.")
+    })
+  })
+
+  context("Form Submission WITH hand", () => {
+
+    it("Input empty SPACE & no score", () => {
```
Comment (dan-dekel) [inline cypress/integration/inputForm.spec.js]: Excellent test case

File: cypress/integration/init.spec.js [hunk 6]
```diff
@@ -68,5 +74,9 @@ describe("Initialise new game instance", () => {
     cy.get('[cy-data="gameScoreList"] li')
       .should("have.length", 1)
       .and("contain", "No history available")
+      .and(() => {
+        expect(localStorage.getItem('historyScores')).to.exist
+        expect(localStorage.getItem('historyScores')).to.contain([]) // to.equal does not work here!
```
Comment (dan-dekel) [inline cypress/integration/init.spec.js]: See above comment

File: cypress/integration/inputForm.spec.js [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+      // Action - no click on new hand
+      // action - click submit button
+      cy.get('[cy-data="submitBtn"]').should("exist").click()
+
+      // Assert - error mesage: You must generate a hand first
+      // Assert - error message: show yellow
+      cy.get('[cy-data="alertTextDisplay"]')
+        .should("be.visible")
+        .contains("You must generate a hand first.")
+      cy.get('[cy-data="alertDisplay"]')
```
Comment (dan-dekel) [inline cypress/integration/inputForm.spec.js]: Interesting, my tests seem to fail in this particular case where cypress can't find the element.. Maybe something to ask in the general-test slack channel?
You could probably use the id of the element for now

File: cypress/integration/init.spec.js (hunk 3 above)
Comment (rachelwong) [reply cypress/integration/init.spec.js]: Done

File: cypress/integration/init.spec.js (hunk 6 above)
Comment (rachelwong) [reply cypress/integration/init.spec.js]: done

File: cypress/integration/inputForm.spec.js (hunk 4 above)
Comment (rachelwong) [reply cypress/integration/inputForm.spec.js]: done

File: cypress/integration/inputForm.spec.js (hunk 7 above)
Comment (rachelwong) [reply cypress/integration/inputForm.spec.js]: done

File: cypress/integration/init.spec.js (hunk 2 above)
Comment (rachelwong) [reply cypress/integration/init.spec.js]: noted and implemented

**Pull request merged**